### PR TITLE
Set C++ 11

### PR DESCRIPTION
**Pull request by moriarty**
Because @lucbettaieb said he got an email about:

http://build.ros.org/job/Ldev__sparse_bundle_adjustment__ubuntu_xenial_amd64/3/

@mikeferguson I also added myself as a maintainer. I'm taking care of whatever the fetch_ros packages depend on.

### CODE REVIEW

**Comment by mikeferguson** [comment]
2 things:

 * I don't think we want to do it this way. Melodic will pass in a build format of C++14 when building debs -- so I think we want to set C++11 ONLY if no build version is set (otherwise we might end up with weird linking errors on Melodic, where everything else is built in C++14).
 * If you're taking over for Luc, should we remove him from the list of maintainers?

**Comment by mikeferguson** [comment]
Disregard my first comment -- [looks like roscpp is hard coding C++11](https://github.com/ros/ros_comm/blob/melodic-devel/clients/roscpp/CMakeLists.txt#L5) -- so I guess we'll be OK.

Second comment remains.

**Comment by lucbettaieb** [comment]
I don't mind getting the emails, etc, so I'm happy to stay on.  Otherwise, I don't mind.

**Comment by mikeferguson** [comment]
Ok -- I'll let that run through CI -- if it passes on Melodic and Lunar, I'll cut a new release. (I think we should also switch slam_karto to melodic-devel branch for Lunar once this is released)(

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(sparse_bundle_adjustment)
 
+add_compile_options(-std=c++11)
```
Comment (dirk-thomas) [inline CMakeLists.txt:4]: The CMake recommended way is now:

```
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 11)
endif()
```

This works cross platform and allows to override it if necessary.

**Comment by mikeferguson** [comment]
Thanks @dirk-thomas -- I'll update to that

**Comment by mikeferguson** [comment]
One note for posterity: CMAKE_CXX_STANDARD requires CMAKE > 3.1, so old CMakeLists needs to be updated to have new minimum version

**Comment by dirk-thomas** [comment]
CMake >= 3.1 :wink:

**Comment by mikeferguson** [comment]
0.4.2 is out for lunar, coming for melodic momentarily

File: CMakeLists.txt (hunk 1 above)
Comment (moriarty) [reply CMakeLists.txt:4]: Ah I thought I remembered seeing  a recommended way...
I just checked tf and went with the way it was done there.

**Pull request merged**